web/unit_tests/cases/engine.js

```diff
@@ -240,3 +223,82 @@ describe('Unmatched Final Groups', function() {
     runKeyboardTestFromJSON('/engine_tests/ghp_enter.json', {usingOSK: true}, done, assert.equal, testconfig.timeouts.scriptLoad);
   });
 });
+
+// Kept separate to maintain an extra-clean setup for this test.
+describe('Engine - Browser Interactions', function() {
+  this.timeout(testconfig.timeouts.scriptLoad);
+
+  before(function() {
+    fixture.setBase('fixtures');
+  });
+
+  beforeEach(function() {
+    fixture.load("singleInput.html");
+    return setupKMW(null, testconfig.timeouts.scriptLoad);
+  });
+
+  afterEach(function() {
+    fixture.cleanup();
+    teardownKMW();
+  });
+
+  describe('Keyboard Loading', function() {
+    it('Local', function() {
+      this.timeout(testconfig.timeouts.scriptLoad);
+
+      var test_callback = function() {
+        assert.isNotNull(keyman.getKeyboard("lao_2008_basic", "lo"), "Keyboard stub was not registered!");
+        assert.equal(keyman.getActiveKeyboard(), "Keyboard_lao_2008_basic", "Keyboard not set automatically!");
+        keyman.removeKeyboards('lao_2008_basic');
+        assert.equal(keyman.getActiveKeyboard(), '', "Keyboard not removed correctly!");
+      }
+
+      return loadKeyboardFromJSON("/keyboards/lao_2008_basic.json", testconfig.timeouts.scriptLoad)
+        .then(test_callback);
+    });
+
+    it('Automatically sets first available keyboard', function() {
+      this.timeout(2 * testconfig.timeouts.scriptLoad);
+
+      var test_callback = function() {
+        assert.isNotNull(keyman.getKeyboard("lao_2008_basic", "lo"), "Keyboard stub was not registered!");
+        assert.equal(keyman.getActiveKeyboard(), "Keyboard_lao_2008_basic", "Keyboard not set automatically!");
+        keyman.removeKeyboards('lao_2008_basic');
+        assert.equal(keyman.getActiveKeyboard(), '', "Keyboard not removed correctly!");
+      }
+
+      return loadKeyboardFromJSON("/keyboards/lao_2008_basic.json", testconfig.timeouts.scriptLoad, {passive: true})
+        .then(() => {
+          // Because we're loading the keyboard 'passively', KMW's setActiveKeyboard function is auto-called
+          // on the stub-add.  That specific call (for first keyboard auto-activation) is outside of KMW's
+          // current Promise chain, so we can't _directly_ rely on a KMW Promise to test it.
```

Comment:
"That specific call":

https://github.com/keymanapp/keyman/blob/47896d8ed5868f09a3d96d4a71ff4dc60c4011ba/web/source/keyboards/kmwkeyboards.ts#L419-L427

This code block is within a function without a return value, thus we can't currently return a `Promise` that will surface for this unit test.